src/Streamly/Internal/Data/Array.hs

```diff
@@ -239,3 +240,30 @@ streamFold f arr = f (toStream arr)
 {-# INLINE getIndexUnsafe #-}
 getIndexUnsafe :: Array a -> Int -> a
 getIndexUnsafe = indexArray
+
+-- | Truncate the array at the beginning and end as long as the predicate
+-- holds true.
+strip :: (a -> Bool) -> Array a -> Array a
+strip p arr =
+    let lastIndex = length arr - 1
+        indexR = getIndexR lastIndex
+        indexL = getIndexL (min 0 lastIndex)
+    in if indexL > indexR
+       then cloneArray arr 0 0
+       else cloneArray arr indexL (indexR - indexL + 1)
+
+    where
+
+    getIndexR idx
+        | idx < 0 = -1
+        | otherwise =
+            case p (indexArray arr idx) of
+            True ->  getIndexR (idx - 1)
+            False -> idx
+
+    getIndexL idx
+        | idx < 0 = 0
+        | otherwise =
+            case p (indexArray arr idx) of
+            True -> getIndexL (idx + 1)
+            False -> idx
```

Comment:
Won't indexR and indexL crash as we are not checking the bounds of the array?

Reply:
getIndexR idx we are checking the boundry .
 getIndexL idx will not be executed in case of getIndexR returns -1.